runtime/src/consensus/tendermint/verifier.rs

```diff
@@ -282,6 +304,7 @@ impl Verifier {
         let state_root = state_root_from_header(&verified_block.signed_header);
         Ok(ConsensusState::from_protocol(
             self.protocol.clone(),
+            state_root.version + 1,
```

Comment:
Should we move this to `from_protocol`?

Reply:
No because it is specific to the Tendermint consensus backend so IMO it should be determined here. I agree that it is somewhat repetitive here and we could maybe move this into a function.

I've tried to explain this in the `ConsensusState` implementation:
```rust
pub struct ConsensusState {
    // An explicit height field is needed because the relationship between the underlying consensus
    // height and the corresponding state root is a consensus backend implementation detail.
    height: u64,
```